src/Access/AccessEntityIO.cpp

```diff
@@ -85,6 +85,16 @@ String serializeAccessEntity(const IAccessEntity & entity)
     WriteBufferFromOwnString buf;
     for (const ASTPtr & query : queries)
     {
+        if (const User * user = typeid_cast<const User *>(&entity))
+        {
+            if (user->auth_data.getSalt().compare("") != 0)
+            {
+                std::string strSalt = " '";
+                strSalt += user->auth_data.getSalt();
```

Comment:
The string is not escaped.

Reply:
the random salt saving in <UUID>.sql file within quotes, after password hash. which would be appended during user login scenario to the input password.